spec/OpenLineage.json

```diff
@@ -64,6 +64,84 @@
         "schemaURL"
       ]
     },
+    "DatasetEvent": {
+      "type": "object",
+      "properties": {
+        "eventTime": {
+          "description": "the time the event occurred at",
+          "type": "string",
+          "format": "date-time"
+        },
+        "dataset": {
+          "allOf": [
+            { "$ref": "#/$defs/Dataset" }
+          ]
+        },
+        "producer": {
+          "description": "URI identifying the producer of this metadata. For example this could be a git url with a given tag or sha",
+          "type": "string",
+          "format": "uri",
+          "example": "https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client"
+        },
+        "schemaURL": {
+          "description": "The JSON Pointer (https://tools.ietf.org/html/rfc6901) URL to the corresponding version of the schema definition for this RunEvent",
+          "type": "string",
+          "format": "uri",
+          "example": "https://openlineage.io/spec/2-0-0/OpenLineage.json#/$defs/DatasetEvent"
```

Comment:
I'm assuming this is how we decide what type of event this is.

Reply:
yes, the approach has been tested within Marquez PR -> https://github.com/MarquezProject/marquez/pull/2495